src/txdb.cpp

```diff
@@ -145,7 +145,7 @@ size_t CCoinsViewDB::EstimateSize() const
     return db.EstimateSize(DB_COIN, (char)(DB_COIN+1));
 }
 
-CBlockTreeDB::CBlockTreeDB(size_t nCacheSize, bool fMemory, bool fWipe) : CDBWrapper(gArgs.IsArgSet("-blocksdir") ? GetDataDir() / "blocks" / "index" : GetBlocksDir() / "index", nCacheSize, fMemory, fWipe) {
+CBlockTreeDB::CBlockTreeDB(size_t nCacheSize, bool fMemory, bool fWipe) : CDBWrapper(GetDataDir() / "blocks" / "index", nCacheSize, fMemory, fWipe) {
```

Comment:
Didn't you pick the wrong side here?
Shouldn't this be `GetBlocksDir() / "index"` instead? Otherwise it seems it's losing the functionality of `-blocksdir`.

Reply:
> Didn't you pick the wrong side here?

This PR does not introduce any change in behavior.

> Shouldn't this be `GetBlocksDir() / "index"` instead? Otherwise it seems it's losing the functionality of `-blocksdir`.

`-blocksdir` allows to place raw blocks to a large and slow HDD, while it leaves blockindex on a fast SSD.

Ref: #12653

Reply:
Whoops. Your'ee right. Thanks for the simplification, that's quite sneaky. I was reading it the wrong way around.

In the future let's review more carefully for people to use an explicit `if()` statement instead of cramming this kind of logic into an initializer.